addon/mixins/route.js

```diff
@@ -236,42 +251,83 @@ const RouteMixin = Mixin.create({
 
     @private
     @method _loadNextPage
     @param {Ember.ArrayProxy} infinityModel
+    @param {Integer} increment - to increase page by 1 or -1. Default to increase by one page
     @return {Ember.RSVP.Promise} A Promise that resolves the model
    */
-  _loadNextPage(infinityModel) {
+  _loadNextPage(infinityModel, increment = 1) {
     set(infinityModel, '_loadingMore', true);
+    set(this, '_previousScrollHeight', this._calculateHeight(infinityModel));
 
     const modelName = get(infinityModel, '_infinityModelName');
-    const params    = infinityModel.buildParams();
+    const params    = infinityModel.buildParams(increment);
 
     return this._requestNextPage(modelName, params)
       .then(newObjects => this._afterInfinityModel(this)(newObjects, infinityModel))
       .then(newObjects => this._doUpdate(newObjects, infinityModel))
       .then(infinityModel => {
-        infinityModel.incrementProperty('currentPage');
+        if (increment === 1) {
+          // scroll down to load next page
+          infinityModel.incrementProperty('currentPage');
+        } else {
+          let viewportElem = get(infinityModel, '_scrollable') ? document.querySelector(get(infinityModel, '_scrollable')) : window;
+          scheduleOnce('afterRender', this, '_updateScrollTop', { infinityModel, viewportElem });
+          // scrolled up to load previous page
+          infinityModel.decrementProperty('currentPage');
+        }
         set(infinityModel, '_firstPageLoaded', true);
-        const canLoadMore = get(infinityModel, '_canLoadMore');
+        let canLoadMore = get(infinityModel, '_canLoadMore');
         set(infinityModel, 'reachedInfinity', !canLoadMore);
         if (!canLoadMore) { this._notifyInfinityModelLoaded(); }
         return infinityModel;
       }).finally(() => set(infinityModel, '_loadingMore', false));
   },
 
+  /**
+    @private
+    @method _calculateHeight
+    @param {Object} infinityModel
+    @return Integer
+   */
+  _calculateHeight(infinityModel) {
+    let viewportElem = get(infinityModel, '_scrollable') ? document.querySelector(get(infinityModel, '_scrollable')) : window;
+    return get(infinityModel, '_scrollable') ? viewportElem.scrollHeight : viewportElem.innerHeight;
+  },
+
+  /**
+    This method calculates the difference if loadPrevious=true
+    The browser by default will scroll to the top of the element list when the previous page
+    loads.  As a result, we need to scroll back down the page.
+    The math behind this is as follows:
+    (height after loading previous elems) - (old height)
+    So 150px - 100px === 150px
+    178px - 100px = 78px
+    120px - 10px = 110px
+    @private
+    @method _updateScrollTop
+    @return Integer
+   */
+  _updateScrollTop({ infinityModel, viewportElem }) {
+    let scrollDiff = this._calculateHeight(infinityModel) - get(this, '_previousScrollHeight');
+    viewportElem.scrollTop += scrollDiff;
+  },
```

Comment:
@sebastianhelbig Here is the logic for scrolling back down the page after the previous page content has rendered.